avoid relying on obscure bash features; assume python3 is OK already.

arnej27959 opened the pull request:
@esolitos please review

arnej27959 committed:
avoid relying on obscure bash features; assume python3 is OK already.

esolitos commented on .buildkite/prepare.sh:
```diff
@@ -6,20 +6,20 @@ set -x
 mkdir -p "$WORKDIR/artifacts/$ARCH/rpms"
 mkdir -p "$WORKDIR/artifacts/$ARCH/maven-repo"
 
-# TODO(aressem): This should move into the vespaengine/docker-image-build-alma* image
-# Make sure we use the latest python3 version installed and ensure that pip is installed
-# shellcheck shell=bash disable=SC2010
-PYBIN="$(ls /usr/bin/python3* | grep -E "/usr/bin/python3.[0-9]+$" |sort -n -k2 -t.|tail -1)"
-alternatives --set python3 "$PYBIN"
-dnf install -y "$(basename "$PYBIN")"-pip
```
Looks good, but we need to make sure we update the default python in AL9, which still uses 3.9:

```bash
❯ docker run --rm -ti vespaengine/vespa-build-almalinux-9 bash

# which python3
/usr/bin/python3

# ls -l /usr/bin/python3
lrwxrwxrwx 1 root root 9 Sep  2 07:19 /usr/bin/python3 -> python3.9

# ls -l /usr/bin/python3*
lrwxrwxrwx 1 root root     9 Sep  2 07:19 /usr/bin/python3 -> python3.9
-rwxr-xr-x 1 root root 68496 Sep  2 07:20 /usr/bin/python3.9
``` 

_While AL8 is correctly using 3.12_

esolitos commented on .buildkite/prepare.sh:
```diff
@@ -6,20 +6,20 @@ set -x
+    awk -F/ '{print $2}' | sort -u > /tmp/list.public-api-modules.txt
+
 # We disable javadoc for all modules not marked as public API
-for MODULE in $(comm -2 -3 \
-                <(find . -name "*.java" | awk -F/ '{print $2}' | sort -u)
-                <(find . -name "package-info.java" -exec grep -HnE "@(com.yahoo.api.annotations.)?PublicApi.*" {} \; | awk -F/ '{print $2}' | sort -u)); do
```
This is not very obscure, but I 100% grant that your implementation is more readable. 👍🏼 

😺

esolitos committed:
fix: shelcheck globbing

arnej27959 committed:
avoid clashing with other maven-jar-plugin configurations

arnej27959 committed:
remove javadoc-disabling

* after discussion with various people, we do not
  want to write to the source tree here; we will
  check in the README files instead and add some
  consistency checking later if needed.
